### PR TITLE
Make work with PostGIS faster

### PR DESCRIPTION
**Pull request by MorriganR**
If there are network delays between the db server and the QGIS, the number of SQL requests to perform simple operations becomes important. This PR combines several SQL requests and sends them to the server in one packet.

### CODE REVIEW

File: src/providers/postgres/qgspostgresconn.cpp [hunk 1]
```diff
@@ -1364,33 +1360,35 @@ int QgsPostgresConn::PQCancel()
 bool QgsPostgresConn::openCursor( const QString &cursorName, const QString &sql )
 {
   QMutexLocker locker( &mLock ); // to protect access to mOpenCursors
+  QString preStr = "";
```
Comment (elpaso) [inline src/providers/postgres/qgspostgresconn.cpp:1363]: no need to initialize a blank string.
```suggestion
  QString preStr;
```

File: src/providers/postgres/qgspostgresconn.cpp [hunk 2]
```diff
@@ -1364,33 +1360,35 @@ int QgsPostgresConn::PQCancel()
...
+                   arg( preStr, cursorName, !mTransaction ? QString() : QStringLiteral( " WITH HOLD" ), sql ) );
 }
 
 bool QgsPostgresConn::closeCursor( const QString &cursorName )
 {
   QMutexLocker locker( &mLock ); // to protect access to mOpenCursors
-
-  if ( !PQexecNR( QStringLiteral( "CLOSE %1" ).arg( cursorName ) ) )
-    return false;
+  QString postStr = "";
```
Comment (elpaso) [inline src/providers/postgres/qgspostgresconn.cpp:1381]:
```suggestion
  QString postStr;
```

File: src/providers/postgres/qgspostgresconn.cpp [hunk 3]
```diff
@@ -1761,6 +1759,68 @@ void QgsPostgresConn::deduceEndian()
...
+    }
+  }
+
+  if ( errorCounter == 0 && oidStatus == ( oidSelectSet | oidBinaryCursorSet ) )
+  {
+    mSwapEndian = mSwapEndian == ( oidSelect == oidBinaryCursor );
+    return;
+  }
+
+  QgsDebugMsg( QStringLiteral( "Back to old deduceEndian(): PQstatus() - %1, queryCounter = %2, errorCounter = %3" )
```
Comment (elpaso) [inline src/providers/postgres/qgspostgresconn.cpp:1819]: use `QgsDebugMsgLevel` instead

**Review by elpaso — APPROVED** [review]
LGTM, just minor changes requested.

**Comment by MorriganR** [comment]
Done, I also did a little rollback in these lines:
https://github.com/qgis/QGIS/blob/a18613c50ac5c2e3a5da4995ea33f3987926ba60/src/providers/postgres/qgspostgresconn.cpp#L1079-L1083
Thanks

**Pull request merged**